src/ktopt_planning_context.cpp

```diff
@@ -48,6 +48,62 @@ void KTOptPlanningContext::solve(planning_interface::MotionPlanResponse& res)
   const auto joint_model_group = getPlanningScene()->getRobotModel()->getJointModelGroup(getGroupName());
   RCLCPP_INFO_STREAM(getLogger(), "Planning for group: " << getGroupName());
 
+  // extract position and velocity bounds
+  std::vector<double> lower_position_bounds;
+  std::vector<double> upper_position_bounds;
+  std::vector<double> lower_velocity_bounds;
+  std::vector<double> upper_velocity_bounds;
+  std::vector<double> lower_acceleration_bounds;
+  std::vector<double> upper_acceleration_bounds;
+  for (const auto& joint_model : joint_model_group->getActiveJointModels())
+  {
+    const std::string& joint_name = joint_model->getName();
+    const moveit::core::VariableBounds& bounds = joint_model->getVariableBounds()[0];
+
+    lower_position_bounds.push_back(bounds.min_position_);
+    upper_position_bounds.push_back(bounds.max_position_);
+    lower_velocity_bounds.push_back(-bounds.max_velocity_);
+    upper_velocity_bounds.push_back(bounds.max_velocity_);
+    lower_acceleration_bounds.push_back(-bounds.max_acceleration_);
+    upper_acceleration_bounds.push_back(bounds.max_acceleration_);
+  }
+  const int num_positions = plant.num_positions();
+  const int num_velocities = plant.num_velocities();
+
+  // Ensure the bounds have the correct size
+  if (lower_position_bounds.size() != num_positions || upper_position_bounds.size() != num_positions)
+  {
+    lower_position_bounds.resize(num_positions, -1e6);
```

Comment:
Instead of using these hard-coded 1e6 magic numbers? Consider defining them as constants in an anonymous namespace.

```cpp
namespace {
constexpr double kDefaultJointMaxPosition = 1.0e6;
}  // namespace
```

But more importantly, did you find that you needed to pad these vectors? Is this because of the additional joints that are not part of the group?

If that's the case, then you possibly want these extra joints to have big position limits (as you did), but also to use zero vel/accel/jerk limits to ensure they do not move.

Reply:
This was a good catch. On closer inspection, I completely missed that the inactive joints were having their kinematics limits added to the end of each respective (position, vel, accel, jerk) bound vector, instead of being indexed. Corrected now!